bcda/cclf/cclf.go

```diff
@@ -258,26 +242,48 @@ func importCCLF(ctx context.Context, fileMetadata *cclfFileMetadata, importFunc
 	}
 	defer rc.Close()
 	sc := bufio.NewScanner(rc)
+
+	// Open transaction to encompass entire CCLF file ingest.
+	txn, err := db.DB().Begin()
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err == nil {
+			err = importer.flush(ctx)
+		}
+
+		if err != nil {
+			err = txn.Rollback()
+		}
+
+		if err = txn.Commit(); err == nil {
+			successMsg := fmt.Sprintf("Successfully imported %d records from CCLF%d file %s.", importedCount, fileMetadata.cclfNum, fileMetadata)
+			fmt.Println(successMsg)
+			log.Infof(successMsg)
+		}
+	}()
+
 	for sc.Scan() {
 		close := metrics.NewChild(ctx, fmt.Sprintf("importCCLF%d-readlines", cclfFile.CCLFNum))
 		b := sc.Bytes()
 		close()
-		if len(bytes.TrimSpace(b)) > 0 {
-			err = importFunc(cclfFile.ID, b, db)
-			if err != nil {
-				log.Error(err)
-				return err
-			}
-			importedCount++
-			if importedCount%importStatusInterval == 0 {
-				fmt.Printf("CCLF%d records imported: %d\n", fileMetadata.cclfNum, importedCount)
-			}
+
+		if len(bytes.TrimSpace(b)) == 0 {
+			continue
+		}
+
+		err = importer.do(ctx, txn, cclfFile.ID, b)
+		if err != nil {
+			log.Error(err)
+			return err
 		}
-	}
 
-	successMsg := fmt.Sprintf("Successfully imported %d records from CCLF%d file %s.", importedCount, fileMetadata.cclfNum, fileMetadata)
-	fmt.Println(successMsg)
-	log.Infof(successMsg)
+		importedCount++
+		if importedCount%importStatusInterval == 0 {
```

Comment:
Is this still needed, now that we've significantly increased the speed of ingestion?  If we do think its needed, should we increase the `importStatusInterval` to maybe 5000 or 10000?

Reply:
:+1: - I'll bump it up to 5k for now. I think it's nice signal to give us an idea that something is going on the instance.